### PR TITLE
PLAT-10343: Datafeed V2 service implementation

### PR DESCRIPTION
**Pull request by symphony-hong**
**Draft PR for early review without unittest yet**

### Ticket
[PLAT-10343](https://perzoinc.atlassian.net/browse/PLAT-10343)

### Description
Datafeed V2 service implementation.

In the BdkDatafeedConfig class, check if the `datafeed.version` configuration is
specified. If not, the datafeed version equals `v1`

In the ServiceFactory class, check if the datafeed version is configured as `v2`, return
the DatafeedLoopV2 instance. Otherwise, the DatafeedLoopV1 will be used by default.

Implementation the datafeed v2 loop service in DatafeedLoopV2 class

### Dependencies
List the other pull requests that should be merged before/along this one.

### Checklist
- [x] Referenced a ticket in the PR title and in the corresponding section
- [x] Filled properly the description and dependencies, if any
- [x] Unit tests updated or added
- [x] Docstrings added or updated
- [ ] Updated the documentation in [docs folder](../docs)

### CODE REVIEW

File: symphony/bdk/core/service/datafeed/datafeed_loop_v2.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    async def _delete_datafeed(self) -> None:
+        session_token = await self.auth_session.session_token
+        key_manager_token = await self.auth_session.key_manager_token
+        await self.datafeed_api.delete_datafeed(datafeed_id=self.datafeed_id,
+                                                session_token=session_token,
+                                                key_manager_token=key_manager_token)
+
+    async def _recreate_datafeed(self):
+        await self._delete_datafeed()
```
Comment (symphony-elias) [inline symphony/bdk/core/service/datafeed/datafeed_loop_v2.py:90]: Shouldn't we reset `self._ack_id` ?

File: tests/core/config/models/test_bdk_datafeed_config.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 import pytest
 
 
-@pytest.fixture(params=["v1", 25, True])
+@pytest.fixture(params=["v1"])
```
Comment (symphony-elias) [inline tests/core/config/models/test_bdk_datafeed_config.py:6]: You should test with v2 and check it returns the DatafeedLoopV2

Comment (symphony-elias) [reply tests/core/config/models/test_bdk_datafeed_config.py:6]: This was actually to test default value of datafeed version is v1.

Comment (symphony-hong) [reply tests/core/config/models/test_bdk_datafeed_config.py:6]: I don't like to have much of logic in the model class. That's why I move the check to the `ServiceFactory` class which will instantiate the DatafeedLoop depending on the version presented in the config object.
https://github.com/SymphonyPlatformSolutions/symphony-api-client-python/pull/124/files#diff-ef818d9162ea0611c768242c2a4089c29eae83aaa77223175f0f78dbb94af5c7R99-R110

Comment (symphony-hong) [reply tests/core/config/models/test_bdk_datafeed_config.py:6]: I added unittest for the `ServiceFactory` class.

**Review by symphony-elias — APPROVED** [review]

**Review by thibauult — COMMENTED** [review]
No need to modify https://github.com/SymphonyPlatformSolutions/symphony-api-client-python/blob/2.0/docs/configuration.md?

Just to let appears that DF version can be changed through `config.yaml` file

**Review by thibauult — APPROVED** [review]
👌

**Pull request merged**